pkg/config/optimizely_config.go

```diff
@@ -67,89 +106,272 @@ type OptimizelyVariable struct {
-	for _, feature := range features {
+func getSerializedAudiences(conditions interface{}, audiencesByID map[string]entities.Audience) string {
+	operators := map[string]bool{}
+	for _, operator := range mappers.GetDefaultOperators() {
+		operators[operator] = true
+	}
+	var serializedAudience string
 
-		var optimizelyVariableMap = map[string]OptimizelyVariable{}
-		for _, variable := range feature.VariableMap {
-			optimizelyVariableMap[variable.Key] = OptimizelyVariable{Key: variable.Key, ID: variable.ID, Value: variable.DefaultValue, Type: string(variable.Type)}
+	if conditions != nil {
+		var cond string
+		if conditionsList, ok := conditions.([]interface{}); ok {
+			for _, condition := range conditionsList {
+				var subAudience string
+				// Checks if item is list of conditions means it is sub audience
+				switch item := condition.(type) {
+				case []interface{}:
+					subAudience = getSerializedAudiences(item, audiencesByID)
+					subAudience = fmt.Sprintf("(%s)", subAudience)
+				case string:
+					if operators[item] {
+						cond = strings.ToUpper(item)
+					} else {
+						// Checks if item is audience id
+						var audienceName = item
+						if audience, ok := audiencesByID[item]; ok {
+							audienceName = audience.Name
+						}
+						// if audience condition is "NOT" then add "NOT" at start.
+						// Otherwise check if there is already audience id in serializedAudience
+						// then append condition between serializedAudience and item
+						if serializedAudience != "" || cond == "NOT" {
+							if cond == "" {
+								cond = "OR"
+							}
+							if serializedAudience == "" {
+								serializedAudience = fmt.Sprintf(`%s "%s"`, cond, audiencesByID[item].Name)
+							} else {
+								serializedAudience += fmt.Sprintf(` %s "%s"`, cond, audienceName)
+							}
+						} else {
+							serializedAudience = fmt.Sprintf(`"%s"`, audienceName)
+						}
+					}
+				default:
+				}
+				// Had to create a different method to reduce cyclomatic complexity
+				evaluateSubAudience(&subAudience, &serializedAudience, &cond)
+			}
+		}
+	}
+	return serializedAudience
+}
 
+func evaluateSubAudience(subAudience, serializedAudience, cond *string) {
+	// Checks if sub audience is empty or not
+	if *subAudience != "" {
+		if *serializedAudience != "" || *cond == "NOT" {
+			if *cond == "" {
+				*cond = "OR"
+			}
+			if *serializedAudience == "" {
+				*serializedAudience = fmt.Sprintf(`%s %s`, *cond, *subAudience)
+			} else {
+				*serializedAudience += fmt.Sprintf(` %s %s`, *cond, *subAudience)
+			}
+		} else {
+			*serializedAudience += *subAudience
 		}
-		for _, experiment := range feature.FeatureExperiments {
-			experimentVariableMap[experiment.Key] = optimizelyVariableMap
+	}
+}
+
+func getExperimentAudiences(experiment entities.Experiment, audiencesByID map[string]entities.Audience) string {
+	if experiment.AudienceConditions == nil {
+		return ""
+	}
+	return getSerializedAudiences(experiment.AudienceConditions, audiencesByID)
```

Comment:
if getSerializedAudiences returns an empty string when experiment.AudienceConditions is nil then we dont need the first check here and can just use the getSerializedAudiences function.